src/org/labkey/test/util/exp/SampleTypeAPIHelper.java

```diff
@@ -84,6 +85,30 @@ public static Map<String, Integer> getRowIdsForSamples(String containerPath, Str
 
         SelectRowsResponse response = cmd.execute(connection, containerPath);
 
+        String errorMsg = "The sample names returned from the query do not match the sample names sent in.";
+
+        // There have been some TC failures where selectRows is not returning anything when it should. Adding this
+        // logging to help get more logging when/if it happens again.
+        if(response.getRowCount().intValue() == 0)
+        {
+            cmd = new SelectRowsCommand("samples", sampleTypeName);
+            cmd.setColumns(Arrays.asList("Name"));
+
+            SelectRowsResponse responseCount = cmd.execute(connection, containerPath);
+
+            errorMsg = errorMsg + "\n" + String.format("No rows were returned with filter. The sample type '%s' has %d rows.",
+                    sampleTypeName, responseCount.getRowCount().intValue());
+
+            List<String> names = new ArrayList<>();
+            for(Map<String, Object> row : response.getRows())
```

Comment:
Seems like you want to use `responseCount` here not `response` (since it has 0 rows).

Reply:
D"oh! Changed.